### PR TITLE
Create completion page

### LINKED ISSUE #63: Create completion page
- [ ] Create the next page
- [ ] Have a list of elements that are the candidate id and the grade
  - the elements should be clickable, ref #64

### PR DESCRIPTION
**Pull request by amalieeh**
Fix #63 #64 
A simple page is created viewing the grades of each student. The cards are clickable, and the page leads the "candidate page" which is where you can see all the answers and score (which is editable) to the student.

### testing purposes
One check to get the tasks is commented out and the task numbers to be viewed are set manually so it is easier to test (aka you don't need to assess ALL of the tasks to view completion page with grades).

### warning in console
There is one warning in the console: Warning: Expected server HTML to contain a matching `<div>` in `<div>`.
A solutions is probably here: https://localcoder.org/react-16-warning-expected-server-html-to-contain-a-matching-div-in-body, but it took me some time trying to understand so I'll rather prioritise other development.

### CODE REVIEW

File: pages/approval.tsx [hunk 1]
```diff
@@ -72,7 +66,7 @@ function filterAssessments(assessments: AssessmentType[]): ApprovalType[] {
             inconsistentScores: inconsistentValues,
           };
         } else {
-          newAss = { ...assessment, score: '', assessmentId: uuidv4() };
```
Comment (amalieeh) [inline pages/approval.tsx:75]: Not related to this PR, just a general fix.

File: pages/task.tsx [hunk 2]
```diff
@@ -4,13 +4,19 @@ import data from '../data/IT2810Høst2018.json';
...
+  getAssessments,
+  noRemainingAnswers,
+  isAllTasksAssessed,
+} from '../functions/helpFunctions';
 import { useEffect, useState } from 'react';
 
 const Task: NextPage = () => {
   const totalTasks: number =
     data.ext_inspera_candidates[0].result.ext_inspera_questions.length;
-  const taskNumbers: number[] = Array.from(Array(totalTasks).keys());
```
Comment (amalieeh) [inline pages/task.tsx:13]: This is also not directly related to this PR, I just caught som bugs because of it, and so I changed it here. Some changes (removing `+1 ` or `-1`) was made due to this, both here and in `helpFunctions.tsx`.

File: pages/completion.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  } else if (score <= 0.76) {
+    return 'C';
+  } else if (score <= 0.88) {
+    return 'B';
+  } else if (score <= 1) {
+    return 'A';
+  } else {
+    return 'Invalid grade';
+  }
+}
```
Comment (amalieeh) [inline pages/completion.tsx:42]: Could probably made a more generic function, but I wanted to finish it fast 😅

File: functions/helpFunctions.tsx [hunk 4]
```diff
@@ -18,34 +19,24 @@ export interface AnswerType {
...
-      ),
-
-      candidateId: parseInt(candidate.result.ext_inspera_candidateId),
-      maxPoints:
-        candidate.result.ext_inspera_questions[questionNumber - 1]
-          .ext_inspera_maxQuestionScore,
-      taskNumber: questionNumber,
-    });
-  });
-  return textboxData;
```
Comment (amalieeh) [inline functions/helpFunctions.tsx:48]: Also not related to this PR, just an unused function, as we discussed earlier :)

File: functions/helpFunctions.tsx [hunk 5]
```diff
@@ -1,8 +1,9 @@
-import { assignInWith } from 'lodash';
 import { v4 as uuidv4 } from 'uuid';
 import { ApprovalType, AssessmentType } from '../types/Types';
 import data from '../data/IT2810Høst2018.json';
 
+export const taskNums = [1, 2, 3]; // For tesing purposes
```
Comment (amalieeh) [inline functions/helpFunctions.tsx:5]: The list for test purposes :) So all tasks are considered finished when these are finished. 
Aka, the button which leads to completion should be clickable when tasks 1, 2, and 3 are approved.

File: components/candidateTextbox.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+interface approvalTexboxProp {
+  assessment: ApprovalType;
+  setAssessmentScore: (assessment: AssessmentType, newScore: number) => void;
+}
+
+const CandidateTextbox: React.FC<approvalTexboxProp> = (
+  props: approvalTexboxProp
+) => {
+  let inconsistentScoresString = '';
```
Comment (sylvihuynh) [inline components/candidateTextbox.tsx:16]: Is this necessary for the `CandidateBox` when it uses the approved assessments from `localstorage` and there are no inconsistent scores saved there?

File: functions/helpFunctions.tsx [hunk 7]
```diff
@@ -18,34 +19,24 @@ export interface AnswerType {
...
-          .ext_inspera_maxQuestionScore,
-      taskNumber: questionNumber,
-    });
-  });
-  return textboxData;
+  return 0;
+};
+
+export function getSumMaxScoresAllTasks(taskKeys: number[]) : number {
+  const rawdata = data; // should be a getData later
```
Comment (sylvihuynh) [inline functions/helpFunctions.tsx:30]: agree on this comment, but aint got time for that now 🐙

File: pages/candidate.tsx [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+import findIndex from 'lodash/findIndex';
+import cloneDeep from 'lodash/cloneDeep';
+import { taskNums } from '../functions/helpFunctions';
+import Header from '../components/header';
+import CandidateTextbox from '../components/candidateTextbox';
+
+function getAllAssessmentsFromCandidate(candidateId: number): ApprovalType[] {
+  let approvedAssessments: ApprovalType[] = [];
+
+  if (typeof window == 'undefined' || isNaN(candidateId) || candidateId == 0) {
```
Comment (sylvihuynh) [inline pages/candidate.tsx:18]: Nice checks ✔️

File: pages/candidate.tsx [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+    newArr.splice(index, 1, newAssessment);
+
+    setAssessments(newArr);
+  };
+
+  const setAssessmentScore = (
+    assessment: AssessmentType,
+    newScore: number | string
+  ) => {
+    // One can no longer set an empty score
```
Comment (sylvihuynh) [inline pages/candidate.tsx:104]: 💯

File: pages/completion.tsx [hunk 10]
```diff
@@ -0,0 +1,139 @@
+import { NextPage } from 'next';
+import styles from '../styles/Completion.module.css';
+import data from '../data/IT2810Høst2018.json';
+import Link from 'next/link';
+import { Button, Snackbar } from '@mui/material';
+import IconButton from "@mui/material/IconButton";
```
Comment (sylvihuynh) [inline pages/completion.tsx:6]:
```suggestion
```

File: pages/completion.tsx [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+import Grid from '@mui/material/Grid';
```
Comment (sylvihuynh) [inline pages/completion.tsx:7]:
```suggestion

```

File: pages/completion.tsx (hunk 10 above)
Comment (sylvihuynh) [reply pages/completion.tsx:6]: These are unused

File: pages/completion.tsx [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+                </Card>
+              </Link>
+            )
+          )}
+        </div>
+        <div style={{ padding: 20 }}>
+            <Button onClick={() => setOpen(true)} style={{ marginLeft: 10 }} variant="contained">
+              Eksporter karakterer
+            </Button>
+          <Snackbar
```
Comment (sylvihuynh) [inline pages/completion.tsx:123]: This was a smooooth workaround 🔥

**Review by sylvihuynh — APPROVED** [review]
Woow, what job!! +400 code lines 🤓  Just a few quick comments to get some cleaner code 😝

File: components/candidateTextbox.tsx (hunk 6 above)
Comment (amalieeh) [reply components/candidateTextbox.tsx:16]: Excellent point! I thought not, BUT I wanted the boxes to have tasknumber, not candidateID, soooo I just made a new one. Had we been earlier in the project I would have made a more generic solution :)

File: functions/helpFunctions.tsx (hunk 7 above)
Comment (amalieeh) [reply functions/helpFunctions.tsx:30]: 🏇

**Review by sylvihuynh — APPROVED** [review]
